gui/src/lib/gui-theme.ts

```diff
@@ -0,0 +1,34 @@
+import { writable } from 'svelte/store';
+
+export const guiThemeLocalStorageKey = 'io.deref.exo/gui-theme';
+
+function createGuiTheme() {
+  const ls = localStorage.getItem(guiThemeLocalStorageKey) || 'auto';
+
+  const { subscribe, set } = writable(ls);
+
+  const localSyncedSet = (s: string) => {
+    set(s);
+    localStorage.setItem(guiThemeLocalStorageKey, s);
+  };
+
+  return {
+    subscribe,
+    auto: () => localSyncedSet('auto'),
+    light: () => localSyncedSet('light'),
+    dark: () => localSyncedSet('dark'),
+    black: () => localSyncedSet('black'),
+  };
+}
+
+export const guiTheme = createGuiTheme();
+
+export const setGuiTheme = (theme: string) => {
+  for (const option of ['auto', 'light', 'dark', 'black']) {
+    if (theme === option) {
```

Comment:
this add/remove pattern can be simplified to use `toggle`. See https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

```typescript
document.body.classList.toggle(option, theme === option)
```